src/core/qgsclipper.h

```diff
@@ -99,19 +104,36 @@ class CORE_EXPORT QgsClipper
 
     SIP_END
 
+    /**
+     * Trims the given polygon to a rectangular box. Modify the given polygon.
+     *
+     * \param pts polygon as 2D coordinates
+     * \param clipRect clipping rectangle
+     */
     static void trimPolygon( QPolygonF &pts, const QgsRectangle &clipRect );
 
     /**
-     * Takes a linestring and clips it to clipExtent
+     * Trims the given polygon to a pseudo 3D box. Modify the given polygon.
+     *
+     * \param pts polygon as 2D or 3D coordinates
+     * \param clipRect clipping 3D box
+     */
+    static void trimPolygon( QgsLineString &pts, const QgsBox3d &clipRect );
+
+    /**
+     * Takes a \a curve with 3D coordinates and clips it to clipExtent
      * \param curve the linestring
      * \param clipExtent clipping bounds
      * \returns clipped line coordinates
+     * \since QGIS 3.21
```

Comment:
```suggestion
     * \since QGIS 3.22
```

(but see above note about leaving the existing clippedLine method available to avoid an API break)